src/SprykerEco/Zed/Computop/Business/Hook/Mapper/Init/InitPayuCeeSingleMapper.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\Computop\Business\Hook\Mapper\Init;
+
+use Generated\Shared\Transfer\QuoteTransfer;
+use Spryker\Shared\Kernel\Transfer\TransferInterface;
+use SprykerEco\Shared\Computop\ComputopConfig;
+use SprykerEco\Shared\Computop\Config\ComputopApiConfig;
+
+class InitPayuCeeSingleMapper extends AbstractMapper
+{
+    /**
+     * @return string
+     */
+    public function getMethodName(): string
+    {
+        return ComputopConfig::PAYMENT_METHOD_PAYU_CEE_SINGLE;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     * @param \Spryker\Shared\Kernel\Transfer\TransferInterface $computopPaymentTransfer
+     *
+     * @return \Spryker\Shared\Kernel\Transfer\TransferInterface
+     */
+    public function updateComputopPaymentTransfer(
+        QuoteTransfer $quoteTransfer,
+        TransferInterface $computopPaymentTransfer
+    ): TransferInterface {
+        /** @var \Generated\Shared\Transfer\ComputopPayuCeeSinglePaymentTransfer $computopPaymentTransfer */
+        $computopPaymentTransfer = parent::updateComputopPaymentTransfer($quoteTransfer, $computopPaymentTransfer);
+        $computopPaymentTransfer
+            ->setMerchantId($this->config->getMerchantId())
+            ->setAmount($quoteTransfer->getTotals()->getGrandTotal());
+
+        $requestTransfer = $this->createRequestTransfer($computopPaymentTransfer);
```

Comment:
* Is it possible to move lines 41 and 42 before line:36?

Reply:
No because method `generateEncryptedMac` use the `merchantId` and `amount` fields.